cli/tools/installer.rs

```diff
@@ -129,7 +131,20 @@ fn get_installer_root() -> Result<PathBuf, io::Error> {
 }
 
 pub fn infer_name_from_url(url: &Url) -> Option<String> {
-  if let Ok(npm_ref) = NpmPackageReference::from_specifier(url) {
+  // If there's an absolute url with no path, eg. https://my-cli.com
+  // perform a request, and see if it redirects another file instead.
+  let url = url.to_owned();
+  let url = if url.path() == "/" {
+    let rt = Runtime::new().unwrap();
```

Comment:
Creating a new runtime here seems like too much overhead. Maybe we should make either this function async or else resolve the redirected specifier at a higher level?

Reply:
I was thinking about this, and wrote it this way, as it's a single call for an entire lifetime of the installation process.
Making it async is definitely a possibility, but it'll require way more changes, which I thought may obstruct a review of this change.